core/servers/chat/mrc_multiplexer.js

```diff
@@ -235,20 +238,15 @@ exports.getModule = class MrcModule extends ServerModule {
      * Takes an MRC message and parses it into something usable
      */
     parseMessage(line) {
-        const msg = line.split('~');
-        if (msg.length < 7) {
-            return;
-        }
 
-        return {
-            from_user: msg[0],
-            from_site: msg[1],
-            from_room: msg[2],
-            to_user: msg[3],
-            to_site: msg[4],
-            to_room: msg[5],
-            body: msg[6]
-        };
+        const [from_user, from_site, from_room, to_user, to_site, to_room, body ] = line.split('~');
+
+        // const msg = line.split('~');
```

Comment:
If you want your sanity check still, `body` (and maybe others) would simply be undefined if not enough elements were produced from the split.

Reply:
I think it should be ok now - it was there because I didn't have the tcp buffering/chunking in place